### PR TITLE
Move mugs derived code to third_party with proper LICENSE

### PR DESCRIPTION
**Pull request by paulb777**

### CODE REVIEW

File: Example/Firebase.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2722,7 +2722,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n";
+			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
```
Comment (schmidt-sebastian) [inline Example/Firebase.xcodeproj/project.pbxproj]: I saw a similar diff in a different PR and it was due to an upgraded CocoaPod version. Can you make sure that you are using the latest available CocoaPods?

File: Firebase/Database/third_party/FImmutableSortedDictionary/LICENSE [hunk 2]
```diff
@@ -0,0 +1,21 @@
+Copyright (c) 2012 Mads Hartmann Jensen
+ +
```
Comment (schmidt-sebastian) [inline Firebase/Database/third_party/FImmutableSortedDictionary/LICENSE]: Can you remove the extra "+" here and below? Thanks!

Comment (paulb777) [reply Firebase/Database/third_party/FImmutableSortedDictionary/LICENSE]: Done. Thanks!

File: Example/Firebase.xcodeproj/project.pbxproj (hunk 1 above)
Comment (paulb777) [reply Example/Firebase.xcodeproj/project.pbxproj]: Yes. This is the output from CocoaPods 1.3.1

**Pull request merged**